db_stress_tool/expected_value.h

```diff
@@ -160,7 +163,51 @@ class PendingExpectedValue {
         orig_value_(orig_value),
         final_value_(final_value) {}
 
+  PendingExpectedValue(const PendingExpectedValue& other)
+      : value_ptr_(other.value_ptr_),
+        orig_value_(other.orig_value_),
+        final_value_(other.final_value_) {
+    other.ClosePendingState();
+  }
+
+  PendingExpectedValue(PendingExpectedValue&& other)
+      : value_ptr_(std::move(other.value_ptr_)),
+        orig_value_(std::move(other.orig_value_)),
+        final_value_(std::move(other.final_value_)) {
+    other.ClosePendingState();
+  }
+
+  PendingExpectedValue& operator=(const PendingExpectedValue& other) {
+    if (this != &other) {
+      other.ClosePendingState();
```

Comment:
Please correct me if I'm wrong here since I am still a C++ n00b. 

In the copy and move constructor, we call ClosePendingState() at the end, but here in copy assignment, we close the state first. Is there a reason for this? Same for move assignment.

Reply:
Ah, I didn't notice this difference, thank you for the question. It was like this more because in the constructor, the `ClosePendingState()` was left at the end after initializer list sets the member variables. I think changing this order should work too.